_rules/aria-role-permitted-j7zzqr.md

```diff
@@ -0,0 +1,252 @@
+
+This rule does not check specifically whether the [explicit role][] is deprecated, or whether it is the special `generic` role that should not be used by authors. These are not conformance requirement, or are checked by other rules.
+
+The roles of `none` or `presentation` are only allowed on certain elements. Therefore this rule also check these elements, even though they are not necessarily [included in the accessibility tree][]. Many of the elements for which `presentation` is not allowed are interactive elements where the role would anyway be ignored due to the [presentational roles conflict resolution][].
+
+### Related rules
+
+- [Role attribute has valid value](https://www.w3.org/WAI/standards-guidelines/act/rules/674b10/proposed/) checks that the value of the `role` attribute exists in ARIA, while this rule checks that it is allowed on the element using it.
+
+### Bibliography
+
+- [Document conformance requirements for use of ARIA attributes in HTML](https://www.w3.org/TR/html-aria/#docconformance)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `a` element with an `href` attribute has an [explicit role][] of `button`, which is allowed.
+
+```html
+<a href="https://www.w3.org/WAI/standards-guidelines/act/rules/" role="button">All ACT rules</a>
+```
+
+#### Passed Example 2
+
+This `h1` element has an [explicit role][] of `tab`, which is allowed
+
+```html
+<h1 role="tab">ACT rules</h1>
+```
+
+#### Passed Example 3
+
+These `hr` elements have an [explicit role][] of `presentation`, which is allowed.
+
+```html
+Fruits:
+<ul>
+	<li>
+		Apple
+		<hr role="presentation" />
+	</li>
+	<li>
+		Banana
+		<hr role="presentation" />
+	</li>
+	<li>Orange</li>
+</ul>
+```
+
+#### Passed Example 4
+
+This `p` element has an [explicit role][] of `generic` which is allowed. Using an explicit role of `generic` for any element is globally not recommended, but not an error by itself.
+
+```html
+<p role="generic">It was a dark and stormy night.</p>
+```
+
+#### Passed Example 5
+
+These `hr` elements have an [explicit role][] of `separator`, which is allowed. Using an explicit role which is the same as the [implicit role][] is not recommended, but not an error by itself.
+
+```html
+Fruits:
+<ul>
+	<li>
+		Apple
+		<hr role="separator" />
+	</li>
+	<li>
+		Banana
+		<hr role="separator" />
+	</li>
+	<li>Orange</li>
+</ul>
+```
+
+#### Passed Example 6
+
+This `h1` element has an [explicit role][] of `tab`, which is allowed. The `btn` token is not a valid role and is therefore ignored.
+
+```html
+<h1 role="btn tab">ACT rules</h1>
+```
+
+#### Passed Example 7
+
+This `dialog` element has an [explicit role][] of `alertdialog`, which is allowed. Even though the `alert` role is not allowed, the first token is a valid role and is therefore used.
+
+```html
+<dialog role="alertdialog alert">This is not right.</dialog>
```

Comment:
I'm not 100% sure about this one. The requirement for it is this:

> The third cell in each row defines the ARIA role values and aria-* attributes which authors MAY specify on the element.

That to me reads that fallback roles apply here too. I don't know that I would want them too. That feels a little on the strict side, but it might be good to put something in the background section about fallback roles, and straight up invalid or abstract roles being out of scope for this rule, even if they do seem like they'd fail the requirement.